Source/WTF/wtf/RetainPtr.h

```diff
@@ -99,7 +99,27 @@ template<typename T> class RetainPtr {
     ~RetainPtr();
 
     void clear();
-    PtrType leakRef() WARN_UNUSED_RETURN;
+
+#ifdef __OBJC__
+    template<typename U = T>
+    std::enable_if_t<std::is_convertible_v<U, id>, PtrType> leakRef() NS_RETURNS_RETAINED WARN_UNUSED_RETURN {
+        static_assert(std::is_same_v<T, U>, "explicit specialization not allowed");
+        return fromStorageType(std::exchange(m_ptr, nullptr));
+    }
+#else
+    template<typename U = T>
+    std::enable_if_t<std::is_same_v<U, id>, PtrType> leakRef() WARN_UNUSED_RETURN {
```

Comment:
Can we use one of the `RETURNS_RETAINED` macros here? Because this does return something retained.

Reply:
I thought neither worked, but I must have been using an earlier iteration of the patch that had a different issue.

`CF_RETURNS_RETAINED` appears to work.  (`NS_RETURNS_RETAINED` won't work because that is only defined in Foundation headers, which aren't available in C++-only sources.)